starknet_py/net/gateway_client.py

```diff
@@ -406,3 +411,21 @@ def get_block_identifier(
         return {"blockNumber": block_number}
 
     return {"blockNumber": "pending"}
+
+
+def _get_payload(
+    tx: Union[StarknetTransaction, List[StarknetTransaction]]
+) -> Union[List, Dict]:
+    type_to_schema = {
+        TransactionType.DECLARE: DeclareSchema(),
+        TransactionType.DEPLOY_ACCOUNT: DeployAccountSchema(),
+        TransactionType.INVOKE_FUNCTION: InvokeSchema(),
+    }
+
+    if isinstance(tx, List):
+        return [
+            type_to_schema[transaction.tx_type()].dump(obj=transaction)
+            for transaction in tx
+        ]
+
+    return type_to_schema[tx.tx_type()].dump(obj=tx)
```

Comment:
Wouldn't maybe something like `marshmallow_dataclass.class_schema(type(tx))` work?

Reply:
Maybe, but is it efficient? Do we want to create it every time?